services/cosmvm/src/node-setup/setup_node.star

```diff
@@ -0,0 +1,152 @@
+        extract={
+            "status":".result.status"
+        }
+    )
+   
+    result = plan.wait(service_name="cosmos",recipe=post_request,field="code",assertion="==",target_value=200)
+
+    return result
+
+def get_last_block(plan):
+
+    post_request = PostHttpRequestRecipe(
+        
+        port_id="rpc",
+        endpoint=ICON_NODE,
+        content_type="application/java",
+        body='{"jsonrpc": "2.0","id": 1,"method": "getLastBlock"}',
+        extract={
+            "height": ".result.height"
+        }
+    )
+
+    response = plan.wait(service_name="cosmos",recipe=post_request,field="code",assertion="==",target_value=200)
+
+    return response["extract.height"]
+
+def get_btp_network_info(plan,service_name,network_id):
+
+    post_request = PostHttpRequestRecipe(
+        port_id="rpc",
+        endpoint=ICON_NODE,
+        content_type="application/java",
+        body='{ "jsonrpc": "2.0", "method": "btp_getNetworkInfo", "id": 1, "params": { "id": "%s" } }' % network_id,
+        extract={
+            "start_height" : '.result.startHeight',
+        }
+    )
+    result = plan.wait(service_name="cosmos",recipe=post_request,field="code",assertion="==",target_value=200)
+
+    exec_command = ["python","-c","print(hex(int(%s) + 1))" % result["extract.start_height"]]
+    result = plan.exec(service_name="cosmos",recipe=ExecRecipe(exec_command))
+
+    execute_cmd = ExecRecipe(command=["/bin/sh", "-c","echo \"%s\" | tr -d '\n\r'" % result["output"] ])
+    result = plan.exec(service_name="cosmos",recipe=execute_cmd)
+
+    return result["output"]
+
+def get_btp_header(plan,network_id,receipt_height):
+
+    post_request = PostHttpRequestRecipe(
+        port_id="rpc",
+        endpoint=ICON_NODE,
+        content_type="application/java",
+        body='{ "jsonrpc": "2.0", "method": "btp_getHeader", "id": 1, "params": { "networkID": "%s" ,"height": "%s" } }' % (network_id,receipt_height),
+        extract={
+            "header" : '.result',
+        }
+    )
+
+    result = plan.wait(service_name="cosmos",recipe=post_request,field="code",assertion="==",target_value=200)
+
+    command = ExecRecipe(command=["python", "-c","from base64 import b64encode, b64decode; print(b64decode('%s').hex())" % result["extract.header"]])
+
+    first_header_hex = plan.exec(service_name="cosmos",recipe=command)
+
+    execute_cmd = ExecRecipe(command=["/bin/sh", "-c","echo \"%s\" | tr -d '\n\r'" % first_header_hex["output"] ])
+    result = plan.exec(service_name="cosmos",recipe=execute_cmd)
+
+    return result["output"]
+
+def int_to_hex(plan,number):
+
+    exec_command = ["python","-c","print(hex(int(%s)))" % number]
+    result = plan.exec(service_name="cosmos",recipe=ExecRecipe(exec_command))
+
+    execute_cmd = ExecRecipe(command=["/bin/sh", "-c","echo \"%s\" | tr -d '\n\r'" % result["output"] ])
+    result = plan.exec(service_name="cosmos",recipe=execute_cmd)
+
+    return result["output"]
+
+
+
+    
+
+    
```

Comment:
This logic is also already present in the icon, Why again here?

Reply:
updated